### PR TITLE
feat/create2 deployer

### PR DESCRIPTION
**Pull request by ethyla**
- uses keyless create2 deployer

### CODE REVIEW

File: src/BridgeExtensionProxy.sol [hunk 1]
```diff
@@ -4,7 +4,7 @@ pragma solidity ^0.8.20;
 import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
 
 contract BridgeExtensionProxy is TransparentUpgradeableProxy {
```
Comment (jesusholoo) [inline src/BridgeExtensionProxy.sol:6]: why don't just deploy the `TransparentUpgradeableProxy` ?¿

File: src/BridgeExtension.sol [hunk 2]
```diff
@@ -21,7 +21,9 @@ contract BridgeExtension is IBridgeAndCall, IBridgeMessageReceiver, Initializabl
 
     PolygonZkEVMBridgeV2 public bridge;
 
-    constructor() {}
+    constructor() {
+        _disableInitializers();
```
Comment (jesusholoo) [inline src/BridgeExtension.sol]: we could add the `bridge` address as immutable and we save ourselves the initialize function :D ( you can always import the initialize lib if you want to reserve the storage slot for a future tho)

File: script/DeployUtils.s.sol [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+contract DeployInitBridgeAndCall is Script {
+
+    address create2Deployer = 0x4e59b44847b379578588920cA78FbF26c0B4956C;
+
+    function run() external {
+
+        bytes memory create2DeployerTx = hex"f8a58085174876e800830186a08080b853604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf31ba02222222222222222222222222222222222222222222222222222222222222222a02222222222222222222222222222222222222222222222222222222222222222";
+
+        address create2DeployerDeployer = 0x3fAB184622Dc19b6109349B94811493BF2a45362;
```
Comment (jesusholoo) [inline script/DeployUtils.s.sol:15]: more than deployer this is the deployer funder right?

Comment (ethyla) [reply script/DeployUtils.s.sol:15]: No, this address actually deploys.
The funder is the users account.

File: src/BridgeExtension.sol (hunk 2 above)
Comment (ethyla) [reply src/BridgeExtension.sol]: Good idea. Makes it a bit cleaner. I will do it.

File: src/BridgeExtensionProxy.sol (hunk 1 above)
Comment (ethyla) [reply src/BridgeExtensionProxy.sol:6]: Used like this it's just an abstraction that doesn't cost anything. And I think a clear name is helpful.
Also it was like this, see no reason to change it.

**Pull request merged**